### PR TITLE
fix(pit-crew): silence Race Engineer + Radar by default (#515)

### LINKED ISSUE #515: fix(audio/actions): Race Engineer and AI Spotter Controls must be disabled unless a Pit Crew button is placed
## Bug

Race Engineer voice callouts (flags, pit‑readback, damage, pit‑status) play even when the user has placed **no Pit Crew button** on any Stream Deck page and has never explicitly enabled Race Engineer. This makes the feature feel like it can't be turned off — there's no UI for it on the user's deck, yet the audio fires.

`AI Spotter Controls` also has no master gate today: any user who places one of those buttons can trigger it without ever opting into the Pit Crew control surface. The user expectation is that Pit Crew is the **master switch** for both subsystems — no Pit Crew button, no Race Engineer, no AI Spotter.

## Root cause (Race Engineer)

The per‑callout opt‑in pattern documented in `.claude/rules/global-settings.md` ("callout<Polarity><Family><Subject>") works correctly **per callout**, but the dispatch predicates in `packages/audio-scenarios/src/catalog/pit-crew/flag-alerts.ts` (and the equivalent damage / pit‑readback / pit‑status builders) only check the per‑callout flag. They do **not** check the master `raceEngineerEnabled` toggle.

Per‑callout flags default `true` (intentional, for forward‑compat). Master `raceEngineerEnabled` defaults `false`. So with no master gate in the predicate, callouts fire on every event regardless of the master toggle.

Reference predicate (current state, `flag-alerts.ts:58-72`):

```ts
const YELLOW_LOCAL: Scenario = {
  ...flagScenario("yellow-local", ["pool:flag-yellow-local"]),
  when: {
    event: "flag.yellow.raised",
    where: (e) => (e as SimEventOf<"flag.yellow.raised">).data.scope === "local",
    // missing: && getGlobalSettings().raceEngineerEnabled === true
  },
};
```

## Desired behavior

1. **Race Engineer audio is gated by `raceEngineerEnabled`.** All voice scenarios (flag, damage, pit‑readback, pit‑status, pit‑service‑requests, plus radar where applicable) must short‑circuit when the master toggle is `false`. Read live on every event arrival (per the rule's "live gating" pattern), not via re-registration on `onGlobalSettingsChange`, so toggles take effect mid‑session without cutting work in flight.
2. **AI Spotter Controls is gated by a new `spotterEnabled` master toggle.** Add `spotterEnabled` to `GlobalSettingsSchema` (default `false`). The `AI Spotter Controls` action's `onKeyDown` short‑circuits with a debug log when `spotterEnabled` is `false`. Add a "Spotter Toggle" mode to the Pit Crew action (parallel to "Race Engineer Toggle" and "Radar"), and a `spotterEnabledOnStartup` field that's mirrored on startup the same way `raceEngineerEnabledOnStartup` is in `plugin.ts`.
3. **Pit Crew button presence acts as a hard safety.** If no Pit Crew action instance is currently visible on any page across any device (tracked via `onWillAppear`/`onWillDisappear` against a plugin‑level `Set<contextId>`), force `raceEngineerEnabled = false` and `spotterEnabled = false` and suppress audio dispatch regardless of stored values. When the first Pit Crew button appears, restore the user's stored value (do not silently flip to `true` — leave the master toggles at whatever the user last set, since the user must still press the button to enable). When the last Pit Crew button disappears, force them back off.
4. **Defaults stay `false`** for `raceEngineerEnabled`, `raceEngineerEnabledOnStartup`, `radarEnabled`, `radarEnabledOnStartup`, `spotterEnabled`, `spotterEnabledOnStartup`. No change for the per‑callout `calloutEnabled*` defaults — those stay `true` for the existing forward‑compat reason.

## Implementation surface

Code:

- `packages/audio-scenarios/src/catalog/pit-crew/flag-alerts.ts` — wrap each scenario's `where` predicate with `getGlobalSettings().raceEngineerEnabled === true && …`. Apply the same change to every other scenario builder under `packages/audio-scenarios/src/catalog/pit-crew/` (damage, pit‑readback, pit‑status, pit‑service‑requests). Prefer a small wrapper helper next to `wrapFlagScenario` so the master gate is applied in one place.
- `packages/deck-core/src/global-settings.ts` — add `spotterEnabled` and `spotterEnabledOnStartup` (boolean, default `false`, with the same `z.union([z.boolean(), z.string()]).transform(...)` pattern).
- `packages/iracing-actions/src/actions/ai-spotter-controls/ai-spotter-controls.ts` — short‑circuit `onKeyDown` when `spotterEnabled` is `false`. Log at debug level so the user can see why a press did nothing.
- `packages/iracing-actions/src/actions/pit-crew/pit-crew.ts` — add a `Spotter Toggle` mode (mirrors the existing `Race Engineer Toggle` mode). Update icon/title accordingly.
- `packages/iracing-actions/src/actions/pit-crew/pit-crew.ejs` — add the new mode to the mode select; expose `spotterEnabledOnStartup` PI control alongside `raceEngineerEnabledOnStartup`.
- `packages/iracing-plugin-stream-deck/src/plugin.ts` — extend the `startupDefaultsApplied` block (currently lines ~286‑297) to also apply `spotterEnabled = spotterEnabledOnStartup`. Add a plugin‑level Pit Crew presence tracker (a `Set<string>` populated by `onWillAppear` / drained by `onWillDisappear` for the Pit Crew action UUID). When the set transitions empty → non‑empty or non‑empty → empty, broadcast a "presence changed" event the audio dispatch can read live (or expose a `isPitCrewOnDeck()` reader the scenario wrapper consults each event).
- `packages/iracing-plugin-mirabox/src/plugin.ts` — same startup‑defaults and presence wiring (must stay in sync with the Stream Deck plugin per `.claude/rules/plugin-structure.md`).

Tests:

- `packages/audio-scenarios/src/catalog/pit-crew/*.test.ts` — assert no scenario fires when `raceEngineerEnabled` is `false`, even with all `calloutEnabled*` flags `true`. Assert no scenario fires when `isPitCrewOnDeck()` returns `false`.
- `packages/iracing-actions/src/actions/ai-spotter-controls/ai-spotter-controls.test.ts` — assert `onKeyDown` is a no‑op when `spotterEnabled` is `false`.
- `packages/iracing-actions/src/actions/pit-crew/pit-crew.test.ts` — cover the new Spotter Toggle mode.

Documentation / artifacts (per `.claude/rules/build-and-commit.md` issue convention — every artifact below that lists Race Engineer / Pit Crew / Spotter must be updated in the same PR):

- All plugin packages — `iracing-plugin-stream-deck`, `iracing-plugin-mirabox`: registration, manifests if a new mode warrants it, PI templates.
- Website (`@iracedeck/website`) — Pit Crew action page (add Spotter Toggle mode), AI Spotter Controls page (mention the master gate), feature list / action counts unaffected (no new actions). Use the per‑mode documentation format from `tire-service.md` (see `.claude/rules/website-action-docs.md`).
- Action documentation (`docs/`) — Pit Crew docs, AI Spotter Controls docs.
- Skills — `iracedeck-actions` skill (mode list for Pit Crew, gating note on AI Spotter Controls), `iracing-telemetry` skill if any new telemetry is read (none expected).
- Rules — `.claude/rules/global-settings.md` (extend the per‑callout opt‑in section with a "master gate" note explaining that families must check both the master enable AND the per‑subject flag).

## Out of scope

- Adding any Spotter audio playback layer (no `packages/audio-scenarios/src/catalog/spotter/`). `AI Spotter Controls` stays button‑only; the master gate is purely a kill switch on `onKeyDown`.
- Changing the per‑callout default of `true` for `calloutEnabled*` flags. Forward‑compat reasoning still holds.
- Reworking Pit Crew presence detection into a generic plugin‑level tracker for other actions. Scope this to Pit Crew only; generalize later if a second consumer appears.

## Acceptance criteria

- Fresh install with no Pit Crew button placed: zero Race Engineer audio fires for any flag/damage/pit event in iRacing. AI Spotter Controls button presses log a debug "spotter disabled" line and do nothing.
- Fresh install with a Pit Crew button placed and `raceEngineerEnabled` toggled on: Race Engineer audio fires for enabled per‑callout subjects.
- Pit Crew button placed → Race Engineer toggled on → Pit Crew button removed from all pages: audio stops within one event of the last `onWillDisappear`.
- Spotter Toggle mode on a Pit Crew button: pressing it flips `spotterEnabled`. AI Spotter Controls obeys the new value without restart.
- All test files above pass under `pnpm test`.
- `pnpm build` and `pnpm lint:fix` clean.

### PR DESCRIPTION
**Pull request by niklam**
## Related Issue

Fixes #515

## What changed?

Race Engineer voice and Radar tick loop fired audibly even when no Pit Crew button was placed on the user's deck. The bus volumes were only muted by the Pit Crew action's `onWillAppear` lifecycle, and dispatch never consulted the master toggles. Three layers of fix:

1. **Rename + migrate** — schema fields renamed to `pitCrewRaceEngineerEnabled` / `pitCrewRadarEnabled` (+ `OnStartup` variants), all default `false`. New `deleteGlobalSettings(keys)` helper in `deck-core` wipes the four pre‑rename keys from persisted storage on first start with this build. Existing users carrying `raceEngineerEnabled=true` / `radarEnabled=true` from any prior build see those values dropped on parse (Zod strips unknown keys) and actively removed from storage.
2. **Master gates at dispatch** — `wrapRaceEngineerMasterGate` in `audio-scenarios` wraps every voice scenario (flag, damage, pit‑readback, pit‑status, fuel/tire/compound/windshield/fast‑repair toggles) as the outermost layer. The radar engine's `getMasterEnabled` closure is consulted on every `radar.changed` arrival AND every scheduled tick. Read live, fires before `attemptFire`, doesn't cut in‑flight clips.
3. **Plugin‑level audio listener** — both plugins now subscribe to `onGlobalSettingsChange` and run `applyRadarVolume` / `applyRadarEnabled` / `applyRaceEngineerAudio` on every change, plus a startup sync. The audio buses + radar engine state always track the master toggles regardless of whether a Pit Crew button is mounted.

Other Pit Crew settings (volumes, voice picker, driver name, callout opt‑ins, audio device) are left intact — the user's choices there carry forward across the migration.

19 new tests added (7 voice master gate + 5 radar master gate + 7 `deleteGlobalSettings`).

## How to test

- [ ] **Fresh install / upgrade**: with no Pit Crew button placed on any deck page, drive into a session and trigger pit‑service toggles, flags, damage. Confirm zero engineer voice and zero radar ticks. The four pre‑rename globals (`raceEngineerEnabled`, `radarEnabled`, `*OnStartup` variants) should be absent from persisted storage after the first start with this build.
- [ ] **Engineer enable**: place a Pit Crew button (mode = Race Engineer Toggle), press once. Confirm engineer voice resumes for the events that previously fired.
- [ ] **Radar enable**: place a Pit Crew button (mode = Radar), press once. Drive next to other cars. Confirm directional tick loop plays.
- [ ] **Disable mid-session**: with engineer / radar enabled, press the toggle to disable. Confirm the next event is silent. Any clip already playing finishes naturally (master gate doesn't cut in‑flight).
- [ ] **Remove buttons mid-session**: with engineer enabled and audible, remove every Pit Crew button from the deck. The action's `onWillDisappear` runs but the global stays `true`; on next plugin restart the on‑startup default (`false`) takes over and the next session is silent until the user re‑adds a button and presses the toggle.
- [ ] **PI On‑startup checkboxes**: in the Pit Crew PI under "On startup", toggle "Race Engineer enabled" and "Radar enabled" on. Restart the plugin. Confirm both runtime toggles come up `true` (and audio fires when relevant events happen). Toggle off, restart, both come up `false`.
- [ ] **Other Pit Crew settings preserved**: confirm Race Engineer Voice, Driver Name, the three volume sliders, audio output device, and every callout opt‑in still hold the values the user set before the upgrade.

## Checklist

- [x] Linked to an approved issue
- [x] New code has unit tests
- [x] Changes registered in all applicable plugins (Stream Deck, Mirabox)
- [x] No unrelated changes included

## Summary by CodeRabbit

* **New Features**
  * Master gate controls added to centrally enable/disable Race Engineer and Radar callouts.

* **Improvements**
  * Settings renamed and reorganized to pitCrew-specific keys; audio state now syncs on startup and on settings changes.
  * Plugin migration cleans up legacy keys and applies new startup defaults.

* **Tests**
  * Added coverage for master-gate behavior, gating edge-cases, and global-settings migration.

* **Chores**
  * Exposed a helper to remove legacy global settings.

### CODE REVIEW

File: packages/audio-scenarios/src/catalog/pit-crew/radar-engine.ts [hunk 1]
```diff
@@ -72,6 +82,12 @@ function startTickLoop(state: ActiveState): void {
   const file = RADAR_AUDIO[state];
   const interval = RADAR_TICK_INTERVALS[state];
   const fire = (): void => {
+    // Master gate (issue #515): even if `enabled` is true and a tick is
+    // already scheduled, refuse to fire when the plugin-wide master
+    // toggle is off. Catches the narrow window where a tick was queued
+    // before `setRadarEnabled(false)` ran but landed after.
+    if (!getMasterEnabled()) return;
```
[the remark this answers is not in the document]
Comment (niklam) [reply packages/audio-scenarios/src/catalog/pit-crew/radar-engine.ts]: Fixed in 4a0c7ca3. Both master-gate checks now call `setVisualState("clear")` before returning, so the icon doesn't stay latched on the last non-clear state when the gate aborts. Channel is intentionally not stopped — keeps the "doesn't cut in-flight" contract that matches the voice master gate. Added two regression tests covering the tick-loop and event-arrival paths.

**Comment by niklam** [comment]
@coderabbitai re: the outside-diff comment on `pit-crew.ts` (lines 472-533 / 625-645) about the last-instance presence force-off — acknowledged, intentionally out of scope for this PR.

The v2 plan was deliberately scoped to rename + migrate + master gates + plugin-level listener; the presence-tracker branch was rejected as scope creep when we landed on this approach. The residual gap (user removes all Pit Crew buttons mid-session while a master toggle is on) is bounded by the on-startup defaults resetting both keys to `false` on next plugin restart.

Tracked as a follow-up: #517.
